types/config.go

```diff
@@ -18,36 +18,37 @@ type SDKConfig struct {
 }
 
 type PocketConfig struct {
-	DataDir                   string `json:"data_dir"`
-	GenesisName               string `json:"genesis_file"`
-	ChainsName                string `json:"chains_name"`
-	EvidenceDBName            string `json:"evidence_db_name"`
-	TendermintURI             string `json:"tendermint_uri"`
-	KeybaseName               string `json:"keybase_name"`
-	RPCPort                   string `json:"rpc_port"`
-	ClientBlockSyncAllowance  int    `json:"client_block_sync_allowance"`
-	MaxEvidenceCacheEntires   int    `json:"max_evidence_cache_entries"`
-	MaxSessionCacheEntries    int    `json:"max_session_cache_entries"`
-	JSONSortRelayResponses    bool   `json:"json_sort_relay_responses"`
-	RemoteCLIURL              string `json:"remote_cli_url"`
-	UserAgent                 string `json:"user_agent"`
-	ValidatorCacheSize        int64  `json:"validator_cache_size"`
-	ApplicationCacheSize      int64  `json:"application_cache_size"`
-	RPCTimeout                int64  `json:"rpc_timeout"`
-	PrometheusAddr            string `json:"pocket_prometheus_port"`
-	PrometheusMaxOpenfiles    int    `json:"prometheus_max_open_files"`
-	MaxClaimAgeForProofRetry  int    `json:"max_claim_age_for_proof_retry"`
-	ProofPrevalidation        bool   `json:"proof_prevalidation"`
-	CtxCacheSize              int    `json:"ctx_cache_size"`
-	ABCILogging               bool   `json:"abci_logging"`
-	RelayErrors               bool   `json:"show_relay_errors"`
-	DisableTxEvents           bool   `json:"disable_tx_events"`
-	Cache                     bool   `json:"-"`
-	IavlCacheSize             int64  `json:"iavl_cache_size"`
-	ChainsHotReload           bool   `json:"chains_hot_reload"`
-	GenerateTokenOnStart      bool   `json:"generate_token_on_start"`
-	LeanPocket                bool   `json:"lean_pocket"`
-	LeanPocketUserKeyFileName string `json:"lean_pocket_user_key_file"`
+	DataDir                    string `json:"data_dir"`
+	GenesisName                string `json:"genesis_file"`
+	ChainsName                 string `json:"chains_name"`
+	EvidenceDBName             string `json:"evidence_db_name"`
+	TendermintURI              string `json:"tendermint_uri"`
+	KeybaseName                string `json:"keybase_name"`
+	RPCPort                    string `json:"rpc_port"`
+	ClientBlockSyncAllowance   int    `json:"client_block_sync_allowance"`
+	ClientSessionSyncAllowance int64  `json:"client_session_sync_allowance"`
```

Comment:
This feature needs to have a feature switch, otherwise weird things can happen when synching from scratch.